### PR TITLE
Improve tests, correctly trigger warnings, and fix output markdown format

### PR DESCRIPTION
**Pull request by tboby**
### WHAT

- Fix the warnings so they actually trigger (#31)
- Fix the latest release output markdown so that it includes correct markdown spacing and list marker
- Port and adapt the integration testing setup from https://github.com/easybuild-org/EasyBuild.PackageReleaseNotes.Tasks (thanks @MangelMaxime !)
- Swap to MSTest (happy to swap back or to any other)

### WHY

Tests are good.

This makes it easier for next steps: probably actually addressing the low-hanging fruit issues, binlog parsing tests, and formatting.

### CODE REVIEW

File: src/Library.fs [hunk 1]
```diff
@@ -47,7 +47,7 @@ type ChangelogExtensions =
     static member Unwrapped(sections: ChangelogSectionCollection) =
         sections
         |> Seq.choose (fun section ->
-            match SemVersion.TryParse section.MarkdownVersion with
+            match SemVersion.TryParse(section.MarkdownVersion, SemVersionStyles.Any) with
```
Comment (baronfel) [inline src/Library.fs:50]: what styles were failing before this change?

**Review by baronfel — APPROVED** [review]

**Comment by baronfel** [comment]
What's the `WIP pipeline` for?

File: src/Library.fs (hunk 1 above)
Comment (tboby) [reply src/Library.fs:50]: None, but `SemVersion.TryParse` without styles provided is deprecated. I couldn't really think of a reason to be stricter. The strictness would prohibit whitespace, prohibit `v` prefixes, and prohibit omiting minor/patch numbers.

**Comment by tboby** [comment]
> What's the `WIP pipeline` for?

It's something fiddly: it's the source you want to set up the nuget.config for your new test folder, but it seemed that if you have it in the pipeline then it can't restore from nuget.org.

See https://github.com/ionide/KeepAChangelog/actions/runs/11494434344/job/31991923809#step:6:206

I thought it was going to be a series of pipeline fixes that I'd squash after but then it worked with one edit...

**Comment by baronfel** [comment]
Ok, so this is good and you have no more WIP for it?

**Comment by tboby** [comment]
> Ok, so this is good and you have no more WIP for it?

Yup!

**Pull request merged**